Sources/Shared/Data Manager/Proxies/Social/SocialPostActivity.m

```diff
@@ -67,11 +73,18 @@ -(void)postActivity:(NSString *)message fileURL:(NSString*)fileURL fileName:(NSS
     manager.router = router;
     
     // Send POST requests for instances of SocialActivity to '/activity.json'
-    [router routeClass:[SocialActivity class] toResourcePath:[self createPath] forMethod:RKRequestMethodPOST];
+    NSString * path = [self createPath];
+    if (space){
+        path = [NSString stringWithFormat:@"%@?identity_id=%@", path, space.spaceId];
+    }
+    [router routeClass:[SocialActivity class] toResourcePath:path forMethod:RKRequestMethodPOST];
     
     // Let's create an SocialActivity
```

Comment:
Move the `SocialActivity` creation after the serialization mapping, so you can add `[activitySimpleMapping mapKeyPath:@"type" toAttribute:@"type"];` in the condition block. If you don't add the serialization mapping for the `type` attribute, it is not present in the request's body.